spectrochempy/analysis/kinetic_utilities.py

```diff
@@ -108,33 +130,35 @@ class ActionMassKinetics(tr.HasTraits):
     _equations = tr.List(tr.Unicode(), help="List of model equations")
     _concentrations = tr.Dict(help="A dictionary of model's species:concentrations")
     _species = tr.List(help="a list of species in this model")
-    _M = Array(help="Stoichiometric matrix M = A+B")
-    _k = Array(help="Arrhenius rate")
+    _A = Array(help="Stoichiometric matrix A (reactants)")
+    _B = Array(help="Stoichiometric matrix B (producs)")
+    _arrhenius = Array(help="Arrhenius rate parameters")
     _T = tr.Union((tr.Float(), tr.Callable()), default_value=298.0, help="Temperature")
 
-    def __init__(self, equations, species, k, T=298.0, **kwargs):
+    def __init__(self, equations, species, arrhenius, T=298.0, **kwargs):
 
-        self._k = k
+        self._arrhenius = arrhenius
         self._T = T
 
         # initialise concentrations, species and equations
         self._concentrations = species
         self._species = list(self._concentrations.keys())
         self._equations = equations
+        self._production_rates = self._write_production_rates()
 
     # ----------------------------------------------------------------------------------
     # Private methods
     # ----------------------------------------------------------------------------------
     @tr.validate("_k")
```

Comment:
The most important here for the validation to occur is to change the string in @tr.validate("_k"). The following func name has no importance except for clarity

Reply:
my mistake !